src/sql/parts/objectExplorer/common/objectExplorerService.ts

```diff
@@ -205,6 +208,29 @@ export class ObjectExplorerService implements IObjectExplorerService {
 		this.sendUpdateNodeEvent(connection, errorMessage);
 	}
 
+	/**
+	 * Gets called when session is disconnected
+	 */
+	public onSessionDisconnected(handle: number, session: sqlops.ObjectExplorerSession) {
```

Comment:
we have a pretty good OE test suite in `objectExplorerService.test.ts`, I'd recommend adding tests to verify this new behavior